SantasBag/SantasBag.Core/Models/Reward.cs

```diff
@@ -2,22 +2,36 @@
 
 public class Reward : BaseEntity
 {
-    public required string Name { get; set; }
+    public string Name { get; set; }
     public string Description { get; set; } = string.Empty;
     public string Image { get; set; } = string.Empty;
-    public required int Cost { get; set; }
-    public required Guid RoomId { get; set; }
+    public decimal Cost { get; set; } = 0;
+    public Guid RoomId { get; set; }
 
-    public Reward(string name, string description, string image, int cost, Guid roomId)
+    private Reward(string name, string description, string image, decimal cost, Guid roomId )
     {
         Id = Guid.NewGuid();
         Name = name;
         Description = description;
         Image = image;
         Cost = cost;
-        RoomId = roomId;
+        //RoomId =roomId;
+        RoomId =Guid.Parse( "00000000-0000-0000-0000-000000000000" );  //временно
     }
 
+    public static (Reward Reward, string Error) Create(string name, string description, string image, decimal cost, Guid roomId)
+    {
+        var error = String.Empty;
+        if (string.IsNullOrEmpty(name))
+        {
```

Comment:
Reward request model should be validated before executing needed endpoint and the validation of name should be executed over there in order to not allow processing invalid data.

Should be refactored.